server/routes/contest.js

```diff
@@ -36,6 +36,26 @@ router.post('/create',
     } catch (err) {
       res.status(422).send({ message: "Improper contest format" });
     }
+  })
+  
+  router.get("/:id",
+  passport.authenticate('jwt', { session: false }),
+  async (req, res) => {
+    try {
+      const { id } = req.params;
+      const contest = await Contest.findOne({
```

Comment:
why did you pass two filters to findOne?

Reply:
When populate fills in the information, it adds in stuff like password which we don't want to return to the client. In this case, createdAt and updatedAt are not necessary for them to know. And they only need to know name, email, profileUrl for our use case. It's a good rule of thumb to limit client's information as much as possible to only what they need to know.